gms/factories/src/main/java/com/linkedin/common/factory/KafkaEventProducerFactory.java

```diff
@@ -23,11 +25,18 @@
   private String kafkaSchemaRegistryUrl;
 
   @Bean(name = "kafkaEventProducer")
-  protected Producer<String, IndexedRecord> createInstance() {
-    Properties props = new Properties();
-    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers);
-    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
-    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class.getName());
+  protected Producer<String, IndexedRecord> createInstance(KafkaProperties properties) {
+    KafkaProperties.Producer producerProps = properties.getProducer();
+
+    producerProps.setKeySerializer(StringSerializer.class);
+    producerProps.setValueSerializer(KafkaAvroSerializer.class);
+
+    // KAFKA_BOOTSTRAP_SERVER has precedence over SPRING_KAFKA_BOOTSTRAP_SERVERS
+    if (kafkaBootstrapServers != null && kafkaBootstrapServers.length() > 0) {
+      producerProps.setBootstrapServers(Arrays.asList(kafkaBootstrapServers.split(",")));
+    } // else we rely on KafkaProperties which defaults to localhost:9092
+
+    Map<String, Object> props = properties.buildProducerProperties();
     props.put(AbstractKafkaAvroSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG, kafkaSchemaRegistryUrl);
```

Comment:
If the default for schema registry URL is `localhost:9092`, we can get rid of the default in line 24 as well?

Reply:
Unfortunately spring-boot has no direct support for schema registry and therefore does not have a default.